daemon/algod/api/server/v2/generated/private/types.go

```diff
@@ -710,11 +710,6 @@ type PostTransactionsResponse struct {
 // ProofResponse defines model for ProofResponse.
 type ProofResponse struct {
 
-	// The type of hash function used to create the proof, must be one of:
-	// * sumhash
-	// * sha512_256
-	Hashtype string `json:"hashtype"`
```

Comment:
How does this break the old API?

Maybe I don't understand what this was compared to the new HashType parameter. It looked like it could be sumhash/sha256/sha512_256.

Like you say, it could be redundant. I'm not sure that's a problem.

Reply:
So I've made these changes to the API:
1. Add a `HashType` request parameter to the `GetProof` endpoint.
2. Remove the `HashType` response parameter since it was never needed to begin with, and also now if the client is asking for a specified `HashType` he can assume that's what he'll get.
Also, I've had a problem with generating the types from the openAPI spec when keeping both `HashType` definitions. I couldn't figure out how to define and reference them both from the same place and decided it wasn't needed as a response anyway.

By breaking the old API I meant the `GetProof` request, so I've made it return the SHA512_256 proof it used to return as a default.

Do you think we should return this hashtype anyway for clarity? Also, do you know how to define this parameter and reference it as both a request and also a response parameter? I couldn't figure it out.

Reply:
one more thing to add here.
I think it was a little bit redundant to add before.  You will never get back the `sumhash` value for example.
if that goes so we left with 2 options (SHA256 or SHA512_256 which the user already asked for a specific one)